external/storm-kafka/src/jvm/storm/kafka/StringScheme.java

```diff
@@ -20,21 +20,23 @@
 import backtype.storm.spout.Scheme;
 import backtype.storm.tuple.Fields;
 import backtype.storm.tuple.Values;
+import backtype.storm.utils.Utils;
 
 import java.io.UnsupportedEncodingException;
+import java.nio.ByteBuffer;
 import java.util.List;
 
 public class StringScheme implements Scheme {
 
     public static final String STRING_SCHEME_KEY = "str";
 
-    public List<Object> deserialize(byte[] bytes) {
+    public List<Object> deserialize(ByteBuffer bytes) {
         return new Values(deserializeString(bytes));
     }
 
-    public static String deserializeString(byte[] string) {
+    public static String deserializeString(ByteBuffer string) {
         try {
-            return new String(string, "UTF-8");
+            return new String(Utils.toByteArray(string), "UTF-8");
```

Comment:
This would create a copy. If the ByteBuffer is array backed, you could use `ByteBuffer.array()` to avoid copy.